pkg/minikube/cluster/cluster.go

```diff
@@ -152,15 +152,15 @@ func UnRegister(api libmachine.API) (bool, error) {
 		return false, err
 	}
 
-	if drivers.MachineInState(host.Driver, state.Running)() {
-		err := registration.UnregisterHostVM(host, RegistrationParameters)
-		if err != nil {
-			return false, err
-		}
-	} else {
-		return true, nil
+	if !drivers.MachineInState(host.Driver, state.Running)() {
+		return true, fmt.Errorf("Unable to verify registration. VM not in running state")
```

Comment:
We can also not return false. It is a tristate (true, false, unknown) since it does not allow only an error to be returned, the true is the least error-prone as we prevent to continue when the information is unclear.